### PR TITLE
Add support for custom transitionConfig

### LINKED ISSUE #1191: Please, can you add android transitioner like ios?
I read the source code and found that ios can use two transition animations by set mode to 'card' or 'modal',but android can't.
```
if (Platform.OS === 'android') {
    // Use the default Android animation no matter if the screen is a modal.
    // Android doesn't have full-screen modals like iOS does, it has dialogs.
    if (prevTransitionProps && (transitionProps.index < prevTransitionProps.index)) {
      // Navigating back to the previous screen
      return FadeOutToBottomAndroid;
    }
    return FadeInFromBottomAndroid;
  } else {
    // iOS and other platforms
    if (isModal) {
      return ModalSlideFromBottomIOS;
    } else {
      return SlideFromRightIOS;
    }
  }
```
So, can you add android transition animations like ios by set mode to 'card' or 'modal',please?

### PR DESCRIPTION
**Pull request by nihgwu**
`transitionConfig` is broken by #984, this fix it

Fixes #1191

### CODE REVIEW

**Comment by grabbou** [comment]
Can we make sure flow tests are passing?

**Comment by grabbou** [comment]
One more thing: Can we make sure this is documented in StackNavigator options, like here: https://github.com/react-community/react-navigation/pull/468/files#diff-d8149c156a733c684c5a21d33a7732daR90 ?

File: docs/api/navigators/StackNavigator.md [hunk 1]
```diff
@@ -85,7 +85,7 @@ Visual options:
   - `screen` - Each screen has a header attached to it and the header fades in and out together with the screen. This is a common pattern on Android.
   - `none` - No header will be rendered.
 - `cardStyle` - Use this prop to override or extend the default style for an individual card in stack.
-- `transitionConfig` - Function to return an object that override default screen transitions.
+- `transitionConfig` - Function to return an object that overrides a default screen transitions.
```
Comment (nihgwu) [inline docs/api/navigators/StackNavigator.md]: a transitions?

**Comment by nihgwu** [comment]
@grabbou 👍

**Comment by MiLeung** [comment]
Where are the docs for transitionConfig?  All I can find is that it is a "Function to return an object that overrides default screen transitions."

What are the properties of the object being returned?

**Comment by nihgwu** [comment]
``` js
  // custom Modal transition animation
  transitionConfig: () => ({
    transitionSpec: {
      duration: 250,
      easing: Easing.out(Easing.poly(4)),
      timing: Animated.timing,
    },
    screenInterpolator: sceneProps => {
      const { layout, position, scene } = sceneProps
      const { index } = scene

      const height = layout.initHeight
      const translateY = position.interpolate({
        inputRange: [index - 1, index, index + 1],
        outputRange: [height, 0, 0],
      })

      const opacity = position.interpolate({
        inputRange: [index - 1, index - 0.99, index],
        outputRange: [0, 1, 1],
      })

      return { opacity, transform: [{ translateY }] }
    },
  }),
```

**Comment by Amerr** [comment]
@nihgwu  can you explain what is  ***screenInterpolator*** and its purpose when its invoked ?

**Comment by nihgwu** [comment]
@Amerr http://www.reactnativediary.com/2016/12/20/navigation-experimental-custom-transition-1.html

**Comment by voidstarfire** [comment]
Where do we get these imports from like "Easing" ?

**Comment by skoob13** [comment]
@voidstarfire do you mean `import { Easing } from 'react-native'`?

**Comment by kalraneeraj24550** [comment]
tansitionConfig working well, if i use stackNavigator, but it is not working when i work with StackRouter, anybody plz help

**Comment by brentvatne** [comment]
please stop spamming @kalraneeraj24550

**Pull request merged**